src/components/NavBar.tsx

```diff
@@ -21,6 +22,31 @@ function classNames(...classes: string[]) {
 }
 
 export default function NavBar() {
+  const [navItems, setNavItems] = useState(navigation);
+
+  useEffect(() => {
+    const path = window.location.pathname;
+    const updatedNavItems = navItems.map(item => {
+      if (item.href === path) {
+        return { ...item, current: true };
+      } else {
+        return { ...item, current: false };
+      }
```

Comment:
Could we do this in 1 line, something like
```
return { ...item, current: item.href === path }
```

Reply:
Yes, we can. I have fixed this in new commit.